main.py

```diff
@@ -55,95 +54,81 @@ def main():
-                out_point = set_timestamp(
-                    "Please specify the trim 'out' point")
-
-                if to_watermark:
-                    # Make the ffmpeg subprocess call.
-                    ffmpeg_call = [
-                        'ffmpeg', '-ss', in_point,
-                        '-to', out_point,
-                        '-i', sys.argv[1],
-                        '-i', watermark_path(aspect_ratio),
-                        '-filter_complex', "[0]yadif=0:-1:1[v];[v][1]overlay",
-                        '-c:v', 'libx264',
-                        '-pix_fmt', 'yuv420p',
-                        '-metadata', 'copyright=Media Archive for Central England',
-                        '-metadata', 'comment=Please contact MACE to license this footage on 01522 837750', 
-                        '-c:a', 'aac',                       
-                        '-report', output_path
-                    ]
-                    print(ffmpeg_call)
-                    subprocess.call(ffmpeg_call)
-                else:
-                    # Make the ffmpeg subprocess call.
-                    ffmpeg_call = [
-                        'ffmpeg', '-ss', in_point,
-                        '-to', out_point,
-                        '-i', input_file,
-                        '-c:v', 'libx264',
-                        '-pix_fmt', 'yuv420p',
-                        '-vf', 'yadif',
-                        '-metadata', 'copyright=Media Archive for Central England',
-                        '-metadata', 'comment=Please contact MACE to license this footage on 01522 837750',                        
-                        '-c:a', 'aac',                        
-                        '-report', output_path
-                    ]
-                    print(ffmpeg_call)
-                    subprocess.call(ffmpeg_call)
-                    
+            ffmpeg_command = create_ffmpeg_command(input_file)
+
+            print("Transcoding with:", " ".join(ffmpeg_command), "\n")
+
+            subprocess.call(ffmpeg_command)
+
+
+def create_ffmpeg_command(input_file):
+    output_path = set_output_path(input_file)
+
+    to_watermark = set_watermark_status()
+    to_trim = set_trim_status()
+
+    aspect_ratio = stream_aspect_ratio(input_file)
+    dar = display_aspect_ratio(aspect_ratio)
+    watermark_file = watermark_path(aspect_ratio)
+
+    ffmpeg_program_call = [
+        "ffmpeg"
+    ]
+
+    input_video_file = [
+        "-i", input_file
+    ]
+
+    watermark_settings = [
+        "-i", watermark_file,
+        "-filter_complex", "[0]yadif=0:-1:1[v];[v][1]overlay"  # overlay filter for supplied png
+    ]
+
+    stream_settings = [
+        # TODO "setdar=dar={0}".format(dar),   # set DAR to aspect ratio
+        "-c:v",     "libx264",          # encode video stream as H.264
+        "-pix_fmt", "yuv420p",          # 4:2:0 chroma subsubsampling
+        "-vf",      "yadif",            # deinterlace video stream
```

Comment:
As above - should probably be `"[0]yadif=0:-1:1[v]"`. (Need to refresh my memory on what that means ...)

Reply:
Removed deinterlacing for the moment in 5b2e721

Like @kieranjol [suggested](https://twitter.com/kieranjol/status/1171756604544430081), it's probably better as an opt-in thing.